packages/product/src/services/product-module-service.ts

```diff
@@ -381,7 +384,7 @@ export default class ProductModuleService<
       productIds,
       sharedContext
     )
-    return JSON.parse(JSON.stringify(products))
+    return serialize(products) as unknown as ProductTypes.ProductDTO[]
```

Comment:
question: Should this live here? We should have mikroorm specific things in the module service, should we? That would put a limitation on the custom DAL

Reply:
My bad, it shouldn't be here but in the dal thanks

Reply:
it should be good [here](https://github.com/medusajs/medusa/pull/4459/commits/f1ce710fb3a30045df7e73a2c152858080583241) now